packages/commerce/frontend/src/lib/valtio-persist.ts

```diff
@@ -0,0 +1,26 @@
+import { proxy, subscribe } from 'valtio'
+
+export function resetStore<
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  T extends Record<string, any>,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  E extends Record<string, any>
+>(store: T, defaults: E) {
+  for (const key in defaults) {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    store[key] = defaults[key]
+  }
+}
+
+export function valtioPersist<TState extends object>(
```

Comment:
We will save the cart store in `localStorage` as well. This will help us to persist the cart store. It will automatically read the given key from `localStorage` and populate the store - if there is data.